src/plugins/resolver/filename.c

```diff
@@ -138,7 +138,35 @@ static char * elektraResolvePasswd (Key * warningsKey)
 	struct passwd pwd;
 	struct passwd * result;
 
-	int s = getpwuid_r (getuid (), &pwd, buf, bufSize, &result);
+	int s = getpwuid_r  (getuid (), &pwd, buf, bufSize, &result);
+
+	if (result == NULL)
+	{
+		elektraFree (buf);
+		if (s != 0)
+		{
+			ELEKTRA_ADD_INSTALLATION_WARNINGF (warningsKey, "Could not retrieve from passwd using getpwuid_r. Reason: %s",
+							   strerror (s));
```

Comment:
Please use strerror_r. Maybe you find also a beautiful way how to use hide this within a macro as it is a very common thing to add warnings/set errors with a string from errno.